### PR TITLE
Mario/2967 refactor bar component

### LINKED ISSUE #2967: refactor cookie, maintenance, and disconnected bar
in app.vue:
```
<CookieBar />
<MaintenanceBar />
<DisconnectedBar />
```
should all use the same `bar` component with the same styles and functionality. 

the container that contains them should be fixed, likely with the following styles:
```
position: fixed;
top: 0;
left: 0;
right: 0;
z-index: 1;
```

if we do this, we'll end up with something that looks like this … 

Screen Shot 2019-09-09 at 11 46 19 AM

so we need to adjust some sidebar styles to figure this one out. 

the whole `#app` element should sit below the fixed container and should only be the height of the space left after the fixed container is accounted for. it should scroll naturally / as expected.

### PR DESCRIPTION
**Pull request by mariopino**
Closes #2967

**Description:**

Refactor cookie, maintenance, and disconnected bar to use the new bar component.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: tests/unit/specs/components/common/__snapshots__/MaintenanceBar.spec.js.snap [hunk 1]
```diff
@@ -1,3 +1,3 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`MaintenanceBar has the expected html structure 1`] = `<!---->`;
+exports[`MaintenanceBar has the expected html structure 1`] = `<div />`;
```
Comment (colw) [inline tests/unit/specs/components/common/__snapshots__/MaintenanceBar.spec.js.snap]: The snapshot would be more useful if some maintenance messages are loaded first.

File: src/components/common/Bar.vue [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    <a class="close">
+      <i class="material-icons close-icon" @click="close(show)">close</i>
+    </a>
+  </div>
+</template>
+
+<script>
+export default {
+  name: `bar`,
+  props: ["show", "type" ],
```
[the remark this answers is not in the document]
Comment (mariopino) [reply src/components/common/Bar.vue]: Prop types are defined in last commit, I also agree that rename 'type' prop is a good idea!

File: src/App.vue [hunk 3]
```diff
@@ -44,4 +46,12 @@ export default {
 
 <style>
 @import "./styles/app.css";
+
+#bar-container {
+  position: fixed;
+  top: 0;
+  left: 0;
+  right: 0;
+  z-index: 1;
```
[the remark this answers is not in the document]
Comment (mariopino) [reply src/App.vue]: Yep, that's fixed in last commit.

File: src/components/common/MaintenanceBar.vue [hunk 4]
```diff
@@ -1,31 +1,22 @@
 <template>
-  <div v-if="maintenance.length > 0">
+  <div>
     <div v-for="(message, index) in maintenance" :key="index">
```
[the remark this answers is not in the document]
Comment (mariopino) [reply src/components/common/MaintenanceBar.vue]: I want to take in account all your appreciations. What do you propose to improve this code?

File: tests/unit/specs/components/common/__snapshots__/MaintenanceBar.spec.js.snap (hunk 1 above)
Comment (mariopino) [reply tests/unit/specs/components/common/__snapshots__/MaintenanceBar.spec.js.snap]: Updated snapshot:

```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintenanceBar has the expected html structure 1`] = `
<div>
  <div>
    <bar-stub
      show="true"
      type="success"
    >
      success test message
    </bar-stub>
  </div>
</div>
`;
```

File: src/components/common/MaintenanceBar.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (colw) [reply src/components/common/MaintenanceBar.vue]: The problem with an index is that it is not unique to an element. If we sorted the list for example, the index would now refer to a different element. The `message` would make for a key.

[the remark this answers is not in the document]
Comment (faboweb) [reply src/components/common/MaintenanceBar.vue]: maintenance messages could have a `id` (UUID) auto generated in postgres. that `id` could serve as key.

[the remark this answers is not in the document]
Comment (mariopino) [reply src/components/common/MaintenanceBar.vue]: Perfect! So added an autoincrement integer field 'id' to maintenance table and used it in the v-for loop:

```
<div v-for="message in maintenance" :key="message.id">
```

**Comment by mariopino** [comment]
All suggestions implemented, ready to merge ...

File: src/components/common/Bar.vue [hunk 5]
```diff
@@ -0,0 +1,91 @@
+<template>
```
Comment (faboweb) [inline src/components/common/Bar.vue]: Nice refactor

**Comment by jbibla** [comment]
yep! awesome refactor.

**Pull request merged**